sdcm/nemesis.py

```diff
@@ -463,6 +463,21 @@ def disrupt_rolling_restart_cluster(self):
         self._set_current_disruption('RollingRestartCluster %s' % self.target_node)
         if not self._is_it_on_kubernetes():
             raise UnsupportedNemesis('RollingRestartCluster is supported only for kubernetes backends')
+
+        # Switch to com.scylladb.auth.SaslauthdAuthenticator
+        if self.cluster.params.get('prepare_saslauthd'):
+            if 'PasswordAuthenticator' in self.cluster.params.get('authenticator'):
+                for node in self.cluster.nodes:
+                    node.remoter.run(
+                        "sed -ie 's/^authenticator:.*/authenticator: com.scylladb.auth.SaslauthdAuthenticator/g' /etc/scylla/scylla.yaml")
+        self.cluster.rollout_restart()
+
+        # Switch back to PasswordAuthenticator
+        if self.cluster.params.get('prepare_saslauthd'):
+            if 'PasswordAuthenticator' in self.cluster.params.get('authenticator'):
+                for node in self.cluster.nodes:
+                    node.remoter.run(
+                        "sed -ie 's/^authenticator:.*/authenticator: PasswordAuthenticator/g' /etc/scylla/scylla.yaml")
```

Comment:
wouldn't it be better to have this in a separated class, like `utils` to be able to call it from other (future) nemesis?

Reply:
Good idea, I will move it to utils.